File: dev.pitlor.SummerCamp/Models/Player.cs
using System.Collections.Immutable;

namespace dev.pitlor.SummerCamp.Models;

public record struct Player(
    string? ConnectionId,
    string Name,
    Color? Color,
    ImmutableList<Card> DrawPile,
    ImmutableList<Card> Hand,
    ImmutableList<Card> DiscardPile,
    int Snackbars,
    int Energy,
    int Path1Progress,
    int Path2Progress,
    int Path3Progress,
    ImmutableList<Badge> Badges);

File: dev.pitlor.SummerCamp.Client/src/models/player.ts
import type { Color } from "./color.ts";
import type { Badge } from "./badge.ts";
import type { Card } from "./card.ts";

export interface Player {
  connectionId?: string;
  name: string;
  color?: Color;
  drawPile: Card[];
  hand: Card[];
  discardPile: Card[];
  snackbars: number;
  path1Progress: number;
  path2Progress: number;
  path3Progress: number;
  badges: Badge[];
}
